### PR TITLE
Add Provider.DiscoveryInfo(...) which retrieves the published IdP configuration info

### PR DESCRIPTION
**Pull request by jimlambrt**
Adding this capability will allow devs to discover and retrieve the provider's published configuration via the well known location of the JSON document.  Using this published configuration, devs can verify that the provider supports the capabilities they need.

### CODE REVIEW

File: oidc/provider.go [hunk 1]
```diff
@@ -660,3 +663,113 @@ func (p *Provider) validRedirect(uri string) error {
...
+	// encode the Claims in a JWT [JWT]. The algorithm RS256 MUST be included.
+	IdTokenSigningAlgsSupported []string `json:"id_token_signing_alg_values_supported"`
+
+	// DisplayValuesSupported (OPTIONAL, omitempty): a list of the display parameter
+	// values that the OpenID Provider supports.
+	DisplayValuesSupported []string `json:"display_values_supported,omitempty"`
+
+	// UILocalesSupported (OPTIONAL, omitempty): Languages and scripts supported
+	// for the user interface
+	UILocalesSupported []string `json:"ui_locales_supported,,omitempty"`
```
Comment (austingebauer) [inline oidc/provider.go:709]: Looks like an additional comma snuck in before `omitempty`?

File: oidc/provider.go [hunk 2]
```diff
@@ -660,3 +663,113 @@ func (p *Provider) validRedirect(uri string) error {
...
+
+	// ClaimsParameterSupported (OPTIONAL, omitempty): Boolean value specifying whether
+	// the OP supports use of the claims parameter, with true indicating support.
+	ClaimsParameterSupported bool `json:"claims_supported"`
```
Comment (austingebauer) [inline oidc/provider.go:713]: Does this also need `omitempty`? Seeing that other fields with `(OPTIONAL, omitempty)` in the comment include it.

**Review by austingebauer — APPROVED** [review]
Just a couple of comments. Otherwise, LGTM!

File: oidc/provider.go (hunk 1 above)
Comment (jimlambrt) [reply oidc/provider.go:709]: ty!

File: oidc/provider.go (hunk 2 above)
Comment (jimlambrt) [reply oidc/provider.go:713]: ty!

**Pull request merged**